expression/aggregation.go

```diff
@@ -76,11 +85,23 @@ func NewAggFunction(funcType string, funcArgs []Expression, distinct bool) Aggre
 
 type aggCtxMapper map[string]*ast.AggEvaluateContext
 
+// AggFunctionMode stands for the aggregation function's mode.
+type AggFunctionMode int
+
+const (
+	// CompleteMode function accepts origin data.
+	CompleteMode AggFunctionMode = iota
```

Comment:
Or if there is only two modes, and we will never add new mode in the future, we can just set `FinalMode` as a bool property in aggFunction.